### PR TITLE
refactor: removing RecipeStatusRegistry and recipeStatusUtils

### PR DESCRIPTION
**Pull request by axel7083**
### What does this PR do?

The RecipeStatus was an interface made of `modelid: number`, `recipeid: number` and `tasks[]`. Its interest was to access quickly the tasks related to given pair of model/recipe.

However, having this _shortcut_ was causing issue, as the tasks was not in a single place, to keep the TaskRegistry as the single source of truth, we needed to remove completely this RecipeStatus and related utility classes. 

The `TaskRegistry` has been enhanced with some new methods, allowing to fetch the requested tasks based on their labels.

### What issues does this PR fix or reference?

Fixes https://github.com/projectatomic/ai-studio/issues/382

### How to test this PR?

- [x] Existing unit tests ensure no regression
- [x] New unit tests has been added

**Manual testing required**
Ensuring no visual regression is important, please test some scenarios
- pulling an application
- restarting application
- deleting manually
- download model(s) from model page

### CODE REVIEW

File: packages/backend/src/managers/modelsManager.ts [hunk 1]
```diff
@@ -183,28 +171,26 @@ export class ModelsManager implements Disposable {
...
-        'model-pulling': model.id,
-      },
-    };
+  async downloadModel(model: ModelInfo, labels?: { [key: string]: string }): Promise<string> {
+    const task: Task = this.taskRegistry.createTask(`Downloading model ${model.name}`, 'loading', {
+      ...labels,
+      'model-pulling': model.id,
+    });
+
+    console.log('Creating model pulling task', task);
```
Comment (lstocchi) [inline packages/backend/src/managers/modelsManager.ts:180]: remove?

File: packages/backend/src/managers/modelsManager.ts [hunk 2]
```diff
@@ -244,7 +230,8 @@ export class ModelsManager implements Disposable {
         }
       }
 
-      this.taskRegistry.set(task); // update task
+      console.log('model manager sending task', task);
```
Comment (lstocchi) [inline packages/backend/src/managers/modelsManager.ts:233]: remove?

File: packages/backend/src/registries/TaskRegistry.ts [hunk 3]
```diff
@@ -16,34 +16,118 @@
...
+   * @param task The task to update.
+   * @throws Error if the task with the specified ID does not exist.
+   */
+  updateTask(task: Task) {
+    if (!this.tasks.has(task.id)) throw new Error(`Task with id ${task.id} does not exist.`);
+    this.tasks.set(task.id, {
+      ...task,
+      state: task.error !== undefined ? 'error' : task.state, // enforce error state when error is defined
+    });
+    console.log('update task notify', this.tasks.get(task.id));
```
Comment (lstocchi) [inline packages/backend/src/registries/TaskRegistry.ts:76]: remove?

File: packages/frontend/src/lib/RecipeDetails.svelte [hunk 4]
```diff
@@ -23,11 +24,17 @@ export let modelId: string;
...
 $: envState = $environmentStates.find((env: EnvironmentState) => env.recipeId === recipeId);
 $: recipe = $catalog.recipes.find(r => r.id === recipeId);
-$: recipeStatus = $recipes.find(r => recipeId === r.recipeId && r.modelId === modelId);
+
+$: filteredTasks = filterByLabel($tasks, {
+  'recipe-id': recipeId,
+  'model-id': modelId,
+});
+
+// $: recipeStatus = $recipes.find(r => recipeId === r.recipeId && r.modelId === modelId);
```
Comment (lstocchi) [inline packages/frontend/src/lib/RecipeDetails.svelte:33]: remove?

**Review by lstocchi — COMMENTED** [review]
LGTM, just some console.log that you forgot i guess

**Comment by axel7083** [comment]
> LGTM, just some console.log that you forgot i guess

@lstocchi yes thanks for noticing! I was too focus avoiding regression I forgot removing the logs I added to debug! 👀

**Pull request merged**